packages/php-wasm/web/vite.config.ts

```diff
@@ -8,144 +8,95 @@ import { viteTsConfigPaths } from '../../vite-extensions/vite-ts-config-paths';
 // eslint-disable-next-line @nx/enforce-module-boundaries
 import { viteIgnoreImports } from '../../vite-extensions/vite-ignore-imports';
 // eslint-disable-next-line @nx/enforce-module-boundaries
+import { vitePreserveLoadersImports } from '../../vite-extensions/vite-preserve-loaders-imports';
+// eslint-disable-next-line @nx/enforce-module-boundaries
 import viteGlobalExtensions from '../../vite-extensions/vite-global-extensions';
 // eslint-disable-next-line @nx/enforce-module-boundaries
 import { getExternalModules } from '../../vite-extensions/vite-external-modules';
 
-export default defineConfig(({ command }) => {
-	return {
-		cacheDir: '../../../node_modules/.vite/php-wasm',
+export default defineConfig({
+	cacheDir: '../../../node_modules/.vite/php-wasm',
 
-		plugins: [
-			viteTsConfigPaths({
-				root: '../../../',
-			}),
-			dts({
-				entryRoot: 'src',
-				tsconfigPath: path.join(__dirname, 'tsconfig.lib.json'),
-				pathsToAliases: false,
-			}),
-			viteIgnoreImports({
-				extensions: ['wasm', 'so', 'dat'],
-			}),
-			/**
-			 * Vite can't extract static asset in the library mode:
-			 * https://github.com/vitejs/vite/issues/3295
-			 *
-			 * This workaround replaces the actual php_5_6.js modules paths used
-			 * in the dev mode with their filenames. Then, the filenames are marked
-			 * as external further down in this config. As a result, the final
-			 * bundle contains literal `import('php_5_6.js')` and
-			 * `import('php_5_6.wasm')` statements which allows the consumers to use
-			 * their own loaders.
-			 *
-			 * This keeps the dev mode working AND avoids inlining 5mb of
-			 * wasm via base64 in the final bundle.
-			 */
+	plugins: [
+		viteTsConfigPaths({
+			root: '../../../',
+		}),
+		dts({
+			entryRoot: 'src',
+			tsconfigPath: path.join(__dirname, 'tsconfig.lib.json'),
+			pathsToAliases: false,
+		}),
+		viteIgnoreImports({
+			extensions: ['wasm', 'so', 'dat'],
+		}),
+		vitePreserveLoadersImports([
 			{
-				name: 'preserve-php-loaders-imports',
-
-				resolveDynamicImport(specifier): string | void {
-					if (
-						command === 'build' &&
-						typeof specifier === 'string' &&
-						specifier.match(/php_\d_\d\.js$/)
-					) {
-						/**
-						 * The ../ is weird but necessary to make the final build say
-						 * import("./php/jspi/php_8_2.js")
-						 * and not
-						 * import("php/jspi/php_8_2.js")
-						 *
-						 * The slice(-3) will ensure the 'php/jspi/'
-						 * portion of the path is preserved.
-						 */
-						return '../' + specifier.split('/').slice(-3).join('/');
-					}
-				},
+				regex: /php_\d_\d\.js$/,
+				/*
+				 * ../ lifts the import located file up to the dist entryRoot
+				 * web/src/lib/get-php-loader-module.ts > web/src/get-php-loader-module.ts
+				 *
+				 * slice(-3) strips the `public` directory from the path
+				 * web/public/php/jspi/php_8_4.js > ./web/php/jspi/php_8_4.js
```

Comment:
```js
"web/public/php/jspi/php_8_4.js".split('/').slice(-3).join('/')
```

yields `php/jspi/php_8_4.js`, not `./web/php/jspi/php_8_4.js`. Let's double check all the examples are accurate